pkg/secretutils/secret_manager.go

```diff
@@ -0,0 +1,140 @@
+package secretutils
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// SecretManager is a type for fetching Secrets whether or not they are in the
+// client cache, labelling so that they will be included in the client cache,
+// and optionally setting an owner reference.
+type SecretManager struct {
+	log       logr.Logger
+	client    client.Client
+	apiReader client.Reader
```

Comment:
SecretManager seems pretty well written. It would be better to add some unit tests by mocking these fields?

Reply:
Maybe it is not easy. There are lots of k8s dependencies.